### ISSUE TITLE
Unusual program behavior on RPi4

### ISSUE THREAD

**Issue by Rak1ta** [Captain-Chaos/WorldPainter#258]
On Raspberry Pi Debian 10 Buster 32 bit, the program only runs with a certain version of java. If something other than 11 is selected, then nothing works. And after installation, after a reboot, an identical additional program shortcut appears in the start menu. 
What might be required to fix this? Left a link to the discussion of the bug in pi-apps. Thanks!

https://github.com/Botspot/pi-apps/issues/1870

**Comment by theofficialgman** [comment]
can confirm the duplicated .desktop icons on all my debian/ubuntu systems.

I'm not sure why @Captain-Chaos you have chosen to copy the .desktop file to /usr/share/applications 
```
ln -s "$I4J_INSTALL_LOCATION/worldpainter.desktop" /usr/share/applications/
```
as well as create one (in a separate folder) using  xdg-desktop-menu

```
xdg-desktop-menu install --mode system "$I4J_INSTALL_LOCATION/.install4j/install4j_1tsntyi-worldpainter.desktop" 2>/dev/null
```
this just results in two .desktop files being created in different directories. choose to make on of them and remove the other if you could please in future debs (this is in the postinst)

**Comment by Captain-Chaos** [comment]
These sound like separate issues; could you make a separate ticket for the duplicate icon please?

Regarding the Java issue: what exactly happens when you try to run WorldPainter with any other version of Java than version 11? Do you have an error message, stack trace or log file?

**Comment by Rak1ta** [comment]
@Captain-Chaos https://github.com/Captain-Chaos/WorldPainter/issues/259

**Comment by Captain-Chaos** [comment]
Thx! I'll take a look. The installer and launcher are generated by Install4J, so I did not write them and have little knowledge of or control over what's in them. But I'll see if I can reproduce this and how to mitigate it.

**Comment by Captain-Chaos** [comment]
Can one of you give me some more context? Is this about installing the regular .deb file on a Raspberry Pi, or is this some kind of bespoke packaging system?

**Comment by Captain-Chaos** [comment]
In the latter case: is it possible that the .tar.gz archive (rather than the installer) might be a better fit?

**Comment by Rak1ta** [comment]
@Captain-Chaos The pi-apps script is used.  Yes, there is a deb. 
https://github.com/Botspot/pi-apps/blob/master/apps/WorldPainter/install

**Comment by theofficialgman** [comment]
> Can one of you give me some more context? Is this about installing the regular .deb file on a Raspberry Pi, or is this some kind of bespoke packaging system?

Normal Deb install. Forcing java 11 is a temporary workaround. No clue why java 16 gives issues on armhf because it works fine on arm64

**Comment by Captain-Chaos** [comment]
Well, I can see the two .desktop files when I install the .deb file myself. No idea why Install4J does that. I have put in a support request with them, perhaps there is something I can do about it.

**Comment by Captain-Chaos** [comment]
Also of course, Java 16 is out of support! Java 17 is the current LTS version. Does it work on Java 17?

**Comment by Captain-Chaos** [comment]
By the way, I took a look at the launch script, and there appears to be a slightly more elegant way to override the JRE: you can create the file `/opt/worldpainter/.install4j/pref_jre.cfg` containing the path to the JRE, rather than having to overwrite the desktop file.

**Comment by theofficialgman** [comment]
> By the way, I took a look at the launch script, and there appears to be a slightly more elegant way to override the JRE: you can create the file `/opt/worldpainter/.install4j/pref_jre.cfg` containing the path to the JRE, rather than having to overwrite the desktop file.

thanks for finding this. I have implemented that now instead of overwritting the .desktop file.

@Captain-Chaos is the support request you opened at install4j publicly accessible? I would like to subscribe to it if it is.

**Comment by Captain-Chaos** [comment]
> @Captain-Chaos is the support request you opened at install4j publicly accessible? I would like to subscribe to it if it is.

No, they don't have a public ticketing system as far as I can see. I just sent an email.

**Comment by Captain-Chaos** [comment]
Alright, it turns out to be a post install script I added myself seven years ago. 😀 I suspect that it was necessary back then, but that in the mean time Install4J gained the same functionality itself so now it is doing it two times. Anyway, I fixed it and just released version 2.10.9, which no longer creates a duplicate desktop file, so I will close this.
